lib/models/ObjectLockConfiguration.ts

```diff
@@ -89,23 +100,18 @@ export default class ObjectLockConfiguration {
      * @param dr - DefaultRetention object containing days or years
      * @return - contains error if parsing failed
      */
-    _parseTime(dr: DefaultRetention): ParsedRetention {
-        if ('Days' in dr && 'Years' in dr) {
+    private parseTime(dr: DefaultRetention): ParsedRetention {
+        if (dr.Days && dr.Years) {
             const msg = 'request xml contains both Days and Years';
             const error = errors.MalformedXML.customizeDescription(msg);
             return { error };
         }
-        const timeType = 'Days' in dr ? 'Days' : 'Years';
+        const timeType = dr.Days ? 'Days' : 'Years';
         if (!dr[timeType] || !dr[timeType][0]) {
             const msg = 'request xml does not contain Days or Years';
```

Comment:
looking at this code, we should push the cleanup a bit further I think: the variable we actually need is `dr[timeType]`, and we can avoid accessing field by name (and later type assertion)

```
if (dr.Days && dr.Years) {
    return 'request xml contains both Days and Years';
}
if (!dr.Days && !dr.Years) {
    return 'request xml does not contain Days or Years';
}
const {timeRaw, timeMax, timeType} = func() {
    if (dr.Days) {
        return {dr.Days, 36500, 'days'};
    } else {
        return {dr.Years, 100, 'years'};
    }
}(); 
const timeValue = Number.parseInt(timeRaw[0], 10);
```